focus-block/src/Components/NewBlockForm/NewBlockForm.js

```diff
@@ -0,0 +1,158 @@
+import React, { Component, Fragment } from 'react';
+import Input from '../Input/Input';
+
+class NewBlockForm extends Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      name: '',
+      timeLimit: '',
+      helpEmail: '',
+      nameIsValid: false,
+      nameErrorShown: false,
+      nameErrorMessage: 'Pick a name for this task',
+      timeLimitIsValid: false,
+      timeLimitErrorShown: false,
+      timeLimitErrorMessage: 'Set a time limit for the task',
+      helpEmailIsValid: true,
+      helpEmailErrorShown: false,
+      helpEmailErrorMessage: 'This is not a valid email address'
+    };
+  }
+
+  nameChangeHandler = (event) => {
+    let nameValue = event.target.value;
+    let nameIsValid = nameValue.length > 0;
+    this.setState({
+      name: nameValue,
+      nameIsValid: nameIsValid
+    });
+  }
+
+  nameBlurHandler = (event) => {
+    this.setState({
+      nameErrorShown: !this.state.nameIsValid
+    });
+  }
+
+  timeLimitChangeHandler = (event) => {
+    let timeLimitValue = event.target.value;
+    let timeLimitIsValid = event.target.value > 0;
+    this.setState({
+      timeLimit: timeLimitValue,
+      timeLimitIsValid: timeLimitIsValid
+    });
+  }
+
+  timeLimitSuggestionClickedHandler = (event) => {
+    let timeLimitValue = event.target.value;
+    this.setState({
+      timeLimit: timeLimitValue,
+      timeLimitIsValid: true,
+      timeLimitErrorShown: false
+    });
+  }
+
+  timeLimitBlurHandler = (event) => {
+    this.setState({
+      timeLimitErrorShown: !this.state.timeLimitIsValid
+    });
+  }
+
+  helpEmailChangeHandler = (event) => {
+    let helpEmailValue = event.target.value;
+    let validContact = helpEmailValue.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
```

Comment:
Lets add this regex as a property, maybe even a global one. This seems to be used in multiple spots.